paragraphs.admin.inc

```diff
@@ -244,6 +244,66 @@ function paragraphs_admin_bundle_form_submit(array $form, array &$form_state) {
 
   $status = paragraphs_bundle_save($bundle);
 
+  // Add permissions if paragraphs_bundle_permissions is enabled and paragraph
+  // is newly created.
+  if (module_exists('paragraphs_bundle_permissions') && $status == SAVED_NEW) {
+    // Define the permission name for viewing paragraph content.
+    $permission_name = 'view paragraph content ' . $bundle->bundle;
+    // Add the permission to the anonymous role.
+    user_role_change_permissions('anonymous', array(
+      $permission_name => TRUE,
+    ));
+    // Define all administrator permissions.
+    $admin_permissions = array('view', 'update', 'delete', 'create');
+    $permissions = array();
+    foreach ($admin_permissions as $permission) {
+      $permission_name = $permission . ' paragraph content ' . $bundle->bundle;
+      $permissions += array(
+        $permission_name => TRUE,
+      );
+    }
+    // Add the permissions to the administrator role.
+    user_role_change_permissions('administrator', $permissions);
+
+    // Define a message informing the user who created the paragraph type that
+    // default permissions have been added.
+    $roles = user_roles();
+    $message = t("Default permissions have been added to the %anonymous role ('View') and %admin role (all) for the %paragraph_type paragraph type. ",
+      array(
+        '%anonymous' => $roles['anonymous'],
+        '%admin' => $roles['administrator'],
+        '%paragraph_type' => $bundle->name,
+      )
+    );
+    // Define part of the message depending on whether the user has permission
+    // to administer permissions or not.
+    if (user_access("administer permissions")) {
+      // Generate a url to the relevant section of the permissions screen.
+      $url_base = 'admin/config/people/permissions';
```

Comment:
This section should be simplified/reworked to use the core permissions filter as well.

Reply:
@yorkshire-pudding If we do end up keeping `paragraphs_bundle_permissions_url` can we use it here as well?